Integrations/Censys/Censys.py

```diff
@@ -1,61 +1,70 @@
+def test_module():
+    url_suffix = "view/ipv4/8.8.8.8"
+    res = requests.get(API_URL + url_suffix, auth=(API_ID, API_SECRET), verify=USE_SSL)
+    if res.status_code == 200:
+        demisto.results('ok')
+    else:
+        demisto.results('test failed')
+
 
-    res = requests.request(method, url + uri, auth=(user, passwd), data=data, verify=USE_SSL)
+def send_request(method, url_suffix, data=None):
+    res = requests.request(method, API_URL + url_suffix, auth=(API_ID, API_SECRET),
+                           data=json.dumps(data), verify=USE_SSL)
+    data = json.loads(res.text)
 
-    if res.status_code >= 400:
-        if res.text.startswith('{'):
-            return '### Error: ' + str(res.status_code) + ': ' + res.json()["error"]
-        else:
-            return '### Error: ' + str(res.status_code)
+    if res.status_code == 404:
+        return None
+    elif res.status_code >= 400:
+        return_error("Error {0}: {1}".format(res.status_code, data["error"].title()))
 
-    return res.text
+    return data
 
 
-apiid = demisto.params()['apiid']
-secret = demisto.params()['secret']
-url = demisto.params()['url']
+def censys_view_command(query, index):
+    url_suffix = 'view/{0}/{1}'.format(index, query)
+    raw = send_request('GET', url_suffix)
+    if raw:
+        demisto.results(raw)
+    else:
+        demisto.results("No view results for {0}.".format(query))
 
-# What happens when the 'Test' button is pressed
-if demisto.command() == 'test-module':
-    res = sendRequest('GET', url, "view/ipv4/8.8.8.8", apiid, secret)
-    if res.startswith('### Error:'):
-        demisto.results({'Type': entryTypes['error'], 'ContentsFormat': 'text', 'Contents': res})
-    elif 'Google' in res:
-        demisto.results('ok')
-else:
-    query = demisto.args()['query']
-    index = demisto.args()['index']
-
-    if demisto.command() == 'cen-view':
-        uri = 'view/' + index + '/' + query
-        method = 'get'
-        data = None
-    elif demisto.command() == 'cen-search':
-        uri = 'search/' + index
-        method = 'POST'
-        if not query.startswith('{'):
-            data = '{ "query" : "' + query + '", "page" : 1 }'
-
-    res = sendRequest(method, url, uri, apiid, secret, data)
-
-    if res.startswith('### Error:'):
-        demisto.results({'Type': entryTypes['error'], 'ContentsFormat': 'text', 'Contents': res})
-        sys.exit(0)
-    elif res.startswith('{'):
-        res = json.loads(res)
-
-    demisto.results(res)
-
-sys.exit(0)
+
+def censys_search_command(query, index):
+    url_suffix = 'search/{0}'.format(index)
+    data = {
+        "query": query,
+        "page": 1
+    }
+    raw = send_request('POST', url_suffix, data)
+    readable = tableToMarkdown("Search results for {0} in {1}".format(query, index), raw["results"])
+    return_outputs(readable, raw)
+
+
+''' EXECUTION CODE '''
+command = demisto.command()
+LOG('command is {0}'.format(command))
+try:
```

Comment:
add handle_proxy() and proxy to the yml file

Reply:
Please check if I've made the changes correctly (in general I'd love to understand more about this option and about fetch incidents, is there any documentation for them?)